set.mm

```diff
@@ -89348,6 +89348,83 @@ being the empty set ( ` A e/ _V ` ).  (Contributed by AV, 6-Aug-2019.) $)
     $( $j usage 'pwfi' avoids 'ax-pow'; $)
   $}
 
+  ${
+    $d A x $.  $d x y z $.  $d u v z $.
+    $( If a set is finite, its converse is as well.  (Contributed by Mario
+       Carneiro, 28-Dec-2014.)  Avoid ~ ax-pow .  (Revised by BTernaryTau,
+       9-Sep-2024.) $)
+    cnvfi $p |- ( A e. Fin -> `' A e. Fin ) $=
+      ( vx vy vz vu vv cv ccnv cfn wcel c0 csn cun wceq eleq1d 0fin cvv cop wex
+      cnveq cnv0 eqeltri cnvun cxp elvv sneq vex cnvsn eqtrdi syl snfi eqeltrdi
+      wi exlimivv sylbi wn crn cdm dfdm4 wne dmsnn0 biimpri necon1bi eqtr3id wb
+      wrel relcnv relrn0 ax-mp sylibr pm2.61i unfi mpan2 eqeltrid a1i findcard2
+      ) BGZHZIJKHZIJCGZHZIJZVTDGZLZMZHZIJZAHZIJBCDAVQKNVRVSIVQKTOVQVTNVRWAIVQVT
+      TOVQWENVRWFIVQWETOVQANVRWHIVQATOVSKIUAPUBWBWGUMVTIJWBWFWAWDHZMZIVTWDUCWBW
+      IIJZWJIJWCQQUDJZWKWLWCEGZFGZRZNZFSESWKEFWCUEWPWKEFWPWIWNWMRZLZIWPWDWOLZNZ
+      WIWRNWCWOUFWTWIWSHWRWDWSTWMWNEUGFUGUHUIUJWQUKULUNUOWLUPZWIKIXAWIUQZKNZWIK
+      NZXAXBWDURZKWDUSWLXEKWLXEKUTWCVAVBVCVDWIVFXDXCVEWDVGWIVHVIVJPULVKWAWIVLVM
+      VNVOVP $.
+    $( $j usage 'cnvfi' avoids 'ax-pow'; $)
+  $}
+
+  ${
+    $d A x y z $.  $d F x y z $.
+    $( A version of ~ fnex for finite sets that does not require Replacement or
+       Power Sets.  (Contributed by Mario Carneiro, 16-Nov-2014.)  (Revised by
+       Mario Carneiro, 24-Jun-2015.) $)
+    fnfi $p |- ( ( F Fn A /\ A e. Fin ) -> F e. Fin ) $=
+      ( vx vy vz cfn wcel wa cres wceq adantr cv wi c0 csn reseq2 eleq1d imbi2d
+      cun a1i wfn fnresdm res0 0fin eqeltri resundi cfv cop wss snfi wfun fnfun
+      funressn syl ssfi sylancr unfi sylan2 eqeltrid findcard2 anabsi7 eqeltrrd
+      expcom a2i ) BAUAZAFGZHZBAIZBFVEVHBJVFABUBKVEVFVHFGZVGBCLZIZFGZMVGBNIZFGZ
+      MVGBDLZIZFGZMZVGBVOELZOZSZIZFGZMZVGVIMCDEAVJNJZVLVNVGWEVKVMFVJNBPQRVJVOJZ
+      VLVQVGWFVKVPFVJVOBPQRVJWAJZVLWCVGWGVKWBFVJWABPQRVJAJZVLVIVGWHVKVHFVJABPQR
+      VNVGVMNFBUCUDUETVRWDMVOFGVGVQWCVQVGWCVQVGHWBVPBVTIZSZFBVOVTUFVGVQWIFGZWJF
+      GVGVSVSBUGUHZOZFGWIWMUIZWKWLUJVEWNVFVEBUKWNABULVSBUMUNKWMWIUOUPVPWIUQURUS
+      VCVDTUTVAVB $.
+    $( $j usage 'fnfi' avoids 'ax-pow' 'ax-rep'; $)
+  $}
+
+  $( If the domain of a one-to-one, onto function is finite, then the domain
+     and range of the function are equinumerous.  This theorem is proved
+     without using the Axiom of Replacement or the Axiom of Power Sets (unlike
+     ~ f1oeng ).  (Contributed by BTernaryTau, 8-Sep-2024.) $)
+  f1oenfi $p |- ( ( A e. Fin /\ F : A -1-1-onto-> B ) -> A ~~ B ) $=
+    ( cfn wcel wf1o cen wbr wfn f1ofn fnfi sylan ancoms f1oen3g sylancom ) ADEZ
+    ABCFZCDEZABGHQPRQCAIPRABCJACKLMABCDNO $.
+  $( $j usage 'f1oenfi' avoids 'ax-pow' 'ax-rep'; $)
+
+  $( If the range of a one-to-one, onto function is finite, then the domain and
+     range of the function are equinumerous.  (Contributed by BTernaryTau,
+     9-Sep-2024.) $)
+  f1oenfirn $p |- ( ( B e. Fin /\ F : A -1-1-onto-> B ) -> A ~~ B ) $=
```

Comment:
There is a possible third statement with antecedent `F e. Fin`.

Reply:
True, but that's already covered by [f1oen3g](https://us.metamath.org/mpeuni/f1oen3g.html), and I don't think this case is important enough to warrant its own theorem.